risc0/zkvm/src/host/recursion/prove/mod.rs

```diff
@@ -411,12 +493,58 @@ impl Prover {
         let (program, control_id) = zkr::join()?;
         let mut prover = Prover::new(program, control_id, opts);
 
-        prover.add_input_digest(&merkle_root);
+        prover.add_input_digest(&merkle_root, DigestKind::Poseidon);
         prover.add_segment_receipt(a, &allowed_ids)?;
         prover.add_segment_receipt(b, &allowed_ids)?;
         Ok(prover)
     }
 
+    /// TODO
+    pub fn new_resolve(
+        cond: &SuccinctReceipt,
+        corr: &SuccinctReceipt,
+        opts: ProverOpts,
+    ) -> Result<Self> {
+        // Construct the Merkle tree of all acceptable recursion predicate control IDs.
+        let hashfn = opts.suite.hashfn.as_ref();
+        let allowed_ids = Self::make_allowed_tree();
+        let merkle_root = allowed_ids.calc_root(hashfn);
+
+        // Load the resolve predicate as a Program and construct the prover.
+        let (program, control_id) = zkr::resolve()?;
+        let mut prover = Prover::new(program, control_id, opts);
+
+        // Load the input values needed by the predicate.
+        // Resolve predicate needs both seals as input, and the journal and assumptions tail digest
+        // to compute the opening of the conditional receipt metadata to the first assumption.
+        prover.add_input_digest(&merkle_root, DigestKind::Poseidon);
+        prover.add_segment_receipt(cond, &allowed_ids)?;
+        prover.add_segment_receipt(corr, &allowed_ids)?;
+
+        let Output {
+            assumptions,
+            journal,
+        } = cond
+            .metadata
+            .output
+            .as_value()
+            .context("cannot resolve conditional receipt with pruned output")?
+            .as_ref()
+            .ok_or(anyhow!("cannot resolve conditional receipt with no output"))?
+            .clone();
+
+        // Unwrap the MaybePruned assumptions list and resolve the corroborated assumption,
+        // removing the head and leaving the tail of the list.
+        let mut assumptions_tail = assumptions
+            .value()
+            .context("cannot resolve conditional receipt with pruned assumptions")?;
+        assumptions_tail.resolve(&corr.metadata.digest())?;
+
+        prover.add_input_digest(&assumptions_tail.digest(), DigestKind::Sha256);
+        prover.add_input_digest(&journal.digest(), DigestKind::Sha256);
+        Ok(prover)
+    }
+
     /// TODO
```

Comment:
Ooops, looks like I missed this one when I ported over the recursion from private.

Reply:
Added a comment to this in 06555ff5a4dcf86c1be0f3444c8692c6bbe69406